pkg/network/server.go

```diff
@@ -682,6 +697,25 @@ func (s *Server) handleTxCmd(tx *transaction.Transaction) error {
 	return nil
 }
 
+// handleP2PNotaryRequestCmd process received P2PNotaryRequest payload.
+func (s *Server) handleP2PNotaryRequestCmd(r *payload.P2PNotaryRequest) error {
+	if !s.chain.P2PSigExtensionsEnabled() {
+		return errors.New("P2PNotaryRequestCMD was received, but P2PSignatureExtensions are disabled")
+	}
+	if err := s.chain.RunWithCurrentState(func() error {
+		return s.notaryRequestPool.VerifyAndPoolPayload(s.chain, r)
+	}); err != nil {
+		return err
+	}
+	go s.broadcastP2PNotaryRequestPayload(r)
```

Comment:
Doing it for each payload is not a good idea, it can be synchronous (or batched as transactions are).

Reply:
Maybe just synchronous for now, we can add batching later if needed.

Reply:
Made it synchronous.